File: core/src/remove/tanks/game/physics/light/LightHandlerPrefab.java
package remove.tanks.game.physics.light;

import com.google.common.base.MoreObjects;
import remove.tanks.game.utility.Prefab;

/**
 * @author Mateusz Długosz
 */
public final class LightHandlerPrefab extends Prefab {
    private final String id;
    private final LightPrefab lightPrefab;

    public LightHandlerPrefab(String id, LightPrefab lightPrefab) {
        this.id = id;
        this.lightPrefab = lightPrefab;
    }

    public String getId() {
        return id;
    }

    public LightPrefab getLightPrefab() {
        return lightPrefab;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("id", id)
                .add("lightPrefab", lightPrefab)
                .toString();
    }
}

File: tools/docs-generator/lib/physics/light/light_handler_prefab.py
class LightHandlerPrefab(object):
    def __init__(self, prefab_id, light_prefab):
        self.id = prefab_id
        self.light_prefab = light_prefab

    def get_id(self):
        return self.id

    def get_light_prefab(self):
        return self.light_prefab

    def __str__(self):
        return "LightHandlerPrefab(id={}, light_prefab={})"\
            .format(self.id, self.light_prefab)
